crates/ethportal-api/src/test_utils/mod.rs

```diff
@@ -1,18 +1,52 @@
 use std::{
-    fs, io,
+    fs::{self, File},
+    io::{self, BufReader},
     path::{Path, PathBuf},
 };
 
+use anyhow::anyhow;
+use serde::de::DeserializeOwned;
+use ssz::Decode;
+
 use self::constants::PORTAL_SPEC_TESTS_SUBMODULE_PATH;
 
 pub mod constants;
+pub mod types;
 
-/// Reads a file from a "portal-spec-tests" submodule.
-pub fn read_file_from_tests_submodule<P: AsRef<Path>>(path: P) -> io::Result<String> {
-    fs::read_to_string(PathBuf::from(PORTAL_SPEC_TESTS_SUBMODULE_PATH).join(path))
+fn portal_spec_tests_path(path: impl AsRef<Path>) -> PathBuf {
+    PathBuf::from(PORTAL_SPEC_TESTS_SUBMODULE_PATH).join(path)
 }
 
 /// Reads bytes from a "portal-spec-tests" submodule.
-pub fn read_bytes_from_tests_submodule<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> {
-    fs::read(PathBuf::from(PORTAL_SPEC_TESTS_SUBMODULE_PATH).join(path))
+pub fn read_binary_portal_spec_tests_file<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> {
+    fs::read(portal_spec_tests_path(path))
+}
+
+/// Reads json file from a "portal-spec-tests" submodule
+pub fn read_json_portal_spec_tests_file<T>(path: impl AsRef<Path>) -> anyhow::Result<T>
+where
+    T: DeserializeOwned,
+{
+    let reader = BufReader::new(File::open(portal_spec_tests_path(path))?);
+    Ok(serde_json::from_reader(reader)?)
+}
+
+/// Reads yaml file from a "portal-spec-tests" submodule
+pub fn read_yaml_portal_spec_tests_file<T>(path: impl AsRef<Path>) -> anyhow::Result<T>
+where
+    T: DeserializeOwned,
+{
+    let reader = BufReader::new(File::open(portal_spec_tests_path(path))?);
+    Ok(serde_yaml::from_reader(reader)?)
+}
+
+/// Reads ssz file from a "portal-spec-tests" submodule
+pub fn read_ssz_portal_spec_tests_file<T: Decode>(path: impl AsRef<Path>) -> anyhow::Result<T> {
+    let bytes = read_binary_portal_spec_tests_file(&path)?;
+    T::from_ssz_bytes(&bytes).map_err(|err| {
+        anyhow!(
+            "Error decoding ssz file: {}. Error: {err:?}",
+            path.as_ref().display()
+        )
+    })
```

Comment:
Isn't this code already defined in `crates/utils/src/submodules.rs` shouldn't we only define it once?